### DISCUSSION TITLE
beta4

### DISCUSSION THREAD

**Discussion by akorotkov** [orioledb/orioledb#325] [Announcements]
The fourth beta release of OrioleDB incorporates the following changes.  Along with bug fixes, this release introduces experimental S3 mode.  See [usage docs](doc/usage.md#s3-database-storage-experimental) for details.

5bf650d Typo fix
faebc36 [skip ci] Further revise description of S3 mode
faebc36 [skip ci] Further revise description of S3 mode
ac8218d Warning when s3 worker cannot find file
35dd610 Prevent writing seq buf when they aren't initialized
c19ad4d Fix read/write concurrency for the same buffer
3b6c7ce orioledb_s3_loader 'endpoint' arg now can be same as orioledb.s3_host
0315775 Document orioledb.s3_desired_size GUC parameter
3b36e76 Small grammar corrections for the docs
fdcfe18 Update docs for s3_loader
58658a6 Add documentation for S3 mode
5adcda6 Do not remove files with locked parts
9dc2055 Fix typo in s3_queue_erase_task()
f88f553 Switch from 5000 to 5002 port for S3 test
4c65d2c Fixes for 10000-partitions test
a7e7e94 pgindent
f870085 Do not perform achiving when S3 mode is disabled
ad61939 Refer pg_hba.conf location in docker postgresql.conf
930ab00 Command-line utility for loading data from S3
1929148 Save S3 mode in control file and check it with the configuration
e47d061 [skip ci] Another attempt to use /etc/postgresql/postgres.conf
adc4d7f Update docs
a43706b Edit config file after initdb
42f608a Further fixes for Dockerfile
82301e5 Change the way of putting config to docker container
b65db4b Attempt to fix valgrind failure
dd314cf issue 300 fix
a3a80c4 issue 297 test added
900a175 issue 257 test
4190fb7 issue 245 fix
32c9eea Drop support of PG 13 and PG 14
4086e30 issue 309 fix
d4bcfc4 More fixes
058bd39 Fix working with "dirty" flag for file parts
036aa49 More bug fix
b6f0239 Fix some bugs in S3 code
d505c0e Harden t.s3_test.S3Test.test_s3_data_eviction test
af18cd6 Fix bug of wrong return value in s3_header_mark_part_loading()
d01775a Fix maintenance of numberOfLoadedParts
d0cbf4f Docker and cppcheck fixes
3d9863d Fixed pg15 build
fd126cc Added libcurl-dev to dockerfiles
4abecbb copied pg_pwrite_zeros from file_utils.c
8a201b6 don't ignore pg_pwrite result
852ff35 Temporary docker option to add s3 if branch contains s3
d6067b0 Missing statement on the test
896f264 Eviction basically works
5c17ed2 pgindent
324e4d0 Improvements
c32528b Lower priority of smaller parts
d58736d pgindent
204814b Buffers synchronization
0f1777f Fix some warnings
af779df WIP: Reading from S3
f0272ae Get rid of separate file for S3 test
b3db966 Removed backup_label
e02cc94 Fixed test_s3_data_dir_load test
204d619 Disabled s3_perform_backup during shutdown checkpoint
17eda29 Limit static analysis versions to 15 and 16
4ca0155 pgindent
6d72374 Fix clang analyzer complaints
6ee3134 Limit supported versions to 15 and 16
fe28589 Fix some warnings
5a90943 Update v16 patchset
65def57 Add an offset for the data
3b84869 Fixes
3c0cf44 Close file descriptors in btree_close_smgr()
77eabd9 Fix small files decoding
3bd59cb Small fixes
ea84d6f Decode small files
7f7efc4 Disable IPv6
c39a5bf Load of postgres files now working
e5c926f [skip ci] WIP OrioledbS3ObjectLoader
d359559 [skip ci] WIP compare checkpointed files
2cddd62 [WIP] [skip ci] S3 test
3ec310b Bulk write small files
c72ee0d Support preloading of WAL files
262b996 [skip ci] Placeholder for loading data pages from S3 in advance
3a9c2a0 Implement S3 mode for OrioleDB
21e7943 Implement checkpoint of non-OrioleDB files to S3
61c65e1 Implement S3 WAL achiever
1451037 Implement the fleet of S3 workers
967585b Implement queue of S3 tasks
ae72f6f Implement S3 requests
6324457 Add docker tags input
6ab02c3 Fixes for docker
df265f0 dockerfile focal job
a6cb5cd Update stopevents_gen.py
e6e4ac8 Update docker_usage.md

<hr /><em>This discussion was created from the release <a href='https://github.com/orioledb/orioledb/releases/tag/beta4'>beta4</a>.</em>

**Comment by lnicola** [comment]
Usage docs at https://github.com/orioledb/orioledb/blob/main/doc/usage.md, for anyone wondering.
